### PR TITLE
issue #344 - properly index Address and HumanName

### PR DESCRIPTION
**Pull request by lmsurpre**
1. add tests in AbstractSearchStringTest
2. modify JDBCParameterBuildingVisitor to pass the tests

previously we were indexing "use" for HumanName and "type" for Addres;
now we index just the string fields

Signed-off-by: Lee Surprenant <lmsurpre@us.ibm.com>

### CODE REVIEW

File: fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/JDBCParameterBuildingVisitor.java [hunk 1]
```diff
@@ -174,64 +174,42 @@ public boolean visit(String elementName, int elementIndex, Uri uri) {
     @Override
     public boolean visit(java.lang.String elementName, int elementIndex, Address address) {
         Parameter p;
-        
-        if (address.getCity() != null) {
+        for (com.ibm.fhir.model.type.String aLine : address.getLine()) {
```
Comment (albertwang-ibm) [inline fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/JDBCParameterBuildingVisitor.java]: Is it safe to drop the  if (address.getLine() != null) check?

File: fhir-persistence/src/test/java/com/ibm/fhir/persistence/search/test/AbstractSearchStringTest.java [hunk 2]
```diff
@@ -117,5 +117,45 @@ public void testSearchString_string_invalidEscaping() throws Exception {
...
+        assertSearchDoesntReturnSavedResource("Address", "both");
+        assertSearchReturnsSavedResource("Address", "4025 S. Miami Blvd., Durham, NC 27703");
+        assertSearchReturnsSavedResource("Address", "4025 S. Miami Blvd.");
+        assertSearchReturnsSavedResource("Address", "Research Triangle Park");
+        assertSearchReturnsSavedResource("Address", "Durham");
+        assertSearchReturnsSavedResource("Address", "NC");
+        assertSearchReturnsSavedResource("Address", "27703");
+        assertSearchReturnsSavedResource("Address", "USA");
+    }
 }
```
Comment (albertwang-ibm) [inline fhir-persistence/src/test/java/com/ibm/fhir/persistence/search/test/AbstractSearchStringTest.java]: I'm wondering if we should add the similar test in server integration test or not...

File: fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/JDBCParameterBuildingVisitor.java (hunk 1 above)
Comment (lmsurpre) [reply fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/JDBCParameterBuildingVisitor.java]: yes.  our model doesn't allow null for repeating elements

File: fhir-persistence/src/test/java/com/ibm/fhir/persistence/search/test/AbstractSearchStringTest.java (hunk 2 above)
Comment (lmsurpre) [reply fhir-persistence/src/test/java/com/ibm/fhir/persistence/search/test/AbstractSearchStringTest.java]: per our discussion on zulip, my thought is to have extensive unit tests and a sampling of integration tests.  i think its good enough just to test this one here

Comment (albertwang-ibm) [reply fhir-persistence/src/test/java/com/ibm/fhir/persistence/search/test/AbstractSearchStringTest.java]: sound good for me

**Pull request merged**